### PR TITLE
Extend the deprecation of `Trainer(resume_from_checkpoint)`

### LINKED ISSUE #10573: Provide a Default Parameter for Fit's Checkpoint Restore Path
## Proposed refactor

Do 1 of 2 options:

1. Do not deprecate `resume_from_checkpoint` from Trainer's initialization as recommended in #9405
2. Provide a public facing property that acts as a default in the case when `ckpt_path` is passed as None to trainer.fit. This property should be accessible after trainer initialization

### Motivation

Our current design pattern is to initialization the Trainer object, and then inject our own logic into the Trainer class, and finally calling trainer.fit(model) without adding any additional arguments to fit(). Moving this argument from trainer's init to fit() will require us to rewrite all of our training scripts.

### Pitch

See Proposed refactor

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Fixes #10573

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

cc @borda @tchaton @rohitgr7

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
LGTM ! Great initiative.

**Review by awaelchli — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]
